_posts/2016-05-16-here-comes-the-telemetry.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+title: "Here comes the telemetry"
+date: 2016-05-16 10:00:00
+author: "@jubianchi"
+categories: news
+---
+
+# Here comes the telemetry
+
+@mageekguy, author of atoum, add the idea to provide a collaborative platform for atoum users so they can share their 
+results and statistics. We did not had much time to work on such things until recently and we are now proud to announce 
+the telemetry platform!
+
+Here it is: [https://dashboard.telemetry.atoum.org](https://dashboard.telemetry.atoum.org).
+
+## Telemetry?
+
+Before getting into details, what telemetry? Here is what [Wikipedia](https://en.wikipedia.org) tell us:
+
+> Telemetry is an automated communications process by which measurements and other data are collected at remote or 
+> inaccessible points and are subsequently transmitted to receiving equipment for monitoring.
+>
+> -- [Telemetry](https://en.wikipedia.org/wiki/Telemetry)
+
+This is exactly what we do with atoum and its telemetry report: we collect some data from your test suites and send them 
+to a global platform so that we can monitor and analyze them.
+
+We chose to make this an opt-in: atoum will not send any data unless you tell it to do so. Enabling this is easy but 
+can't be done by mistake. If you want to send us your data, you will have to add the 
+[`reports-extension`](https://github.com/atoum/reports-extension) and enable the required report.
+
+First, add the new dependency with composer:
+
+```
+$ composer require --dev atoum/reports-extension
+```
+
+THen enable the report in your configuration file, `.atoum.php`:
+
+```
+<?php
+
+use mageekguy\atoum\reports\telemetry;
+use mageekguy\atoum\writers\std;
+
+$script->addDefaultReport();
+
+$telemetry = new telemetry();
+$telemetry->addWriter(new std\out());
+$runner->addReport($telemetry);
+```
+
+Now each time you run tour test suite, results will be sent to the telemetry.
+
+Because the dashboards are publicly accessible, by default everything will be sent anonymously. If you want to let us 
+know who you are and which project is producing those data, you can add the following lines:
+
+```
+<?php
+
+$telemetry->readProjectNameFromComposerJson(__DIR__ . '/composer.json');
+
+// Or
+
+$telemetry->setProjectName('my/project');
+
+// And optionally
+
+$telemetry->sendAnonymousProjectName();
```

Comment:
if we do 

```php
<?php

$telemetry->setProjectName('my/project');
$telemetry->sendAnonymousProjectName();
```

the the report will anonymize the project part only and send something like `my/anon-84c89efabaa6980fa47a75d44800df05` as the project name